File: a-case-for-integration-tests/src/itest/java/com/coderstower/blog/a_case_integration_tests/extension/ApplicationContextCacheValidatorITest.java
package com.coderstower.blog.a_case_integration_tests.extension;

import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Order;
import org.junit.jupiter.api.Test;
import org.springframework.boot.test.system.CapturedOutput;
import org.springframework.test.context.ActiveProfiles;

// Test run last
@Order(Integer.MAX_VALUE)
@ActiveProfiles(value = "newProfile")
class ApplicationContextCacheValidatorITest extends MockedEdgesConfig {

    @Test
    @DisplayName(
            """
                        Validate the amount of cached application context doesn't increase
                    """
    )
    public void validate(CapturedOutput output, ITestHandler iTestHandler) {
        iTestHandler.assertCacheSizeIsLessThanOrEqualTo(output, 9);
    }
}

File: a-case-for-integration-tests/src/itest/java/com/coderstower/blog/a_case_integration_tests/application/ApplicationContextCacheValidatorItest.kt
package com.coderstower.blog.a_case_integration_tests.application

import com.coderstower.blog.a_case_integration_tests.extesion.CachedApplicationContextValidator
import com.coderstower.blog.a_case_integration_tests.extesion.TestApplicationContextExtension
import org.junit.jupiter.api.DisplayName
import org.junit.jupiter.api.Order
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.extension.ExtendWith
import org.springframework.boot.test.system.CapturedOutput
import org.springframework.boot.test.system.OutputCaptureExtension
import org.springframework.test.context.ActiveProfiles

// Test run last
@Order(Integer.MAX_VALUE)
@ExtendWith(TestApplicationContextExtension::class)
@ExtendWith(OutputCaptureExtension::class)
@ActiveProfiles(value = ["newProfile"])
class ApplicationContextCacheValidatorItest : MockedEdgesConfig() {

    @Test
    @DisplayName(
        """
            Validate the amount of cached application context doesn't increase
        """
    )
    fun validate(output: CapturedOutput, cachedApplicationContextValidator: CachedApplicationContextValidator) {
        cachedApplicationContextValidator.assertCacheSizeIsLessThanOrEqualTo(output, 9)
    }
}
